src/ui/Base/QueryBuilder.ts

```diff
@@ -282,8 +282,21 @@ export class QueryBuilder {
 
   /**
    * Specifies an array of request for the MLFacet component.
+   * Can not be used at the same time as [`groupByRequests`]{@link QueryBuilder.groupByRequests}
+   * Use the [`addFacetRequest`]{@link QueryBuilder.addFacetRequest} method to define this value
```

Comment:
Rather than documenting the method to use, I think a better approach would be to make the variable private. This reduces the chance of someone using the class incorrectly. If you need access to the property outside of the class, I would use a getter.

Also, I would initialize to an empty array on 288 to avoid the `if` block on lines 294-296.
To return `undefined`, you could check the length of the array inside the getter. If the length is `0`, then we can return `undefined`.